storage_drivers/azure/api/azure.go

```diff
@@ -131,15 +131,26 @@ type Client struct {
 // NewDriver is a factory method for creating a new SDK interface.
 func NewDriver(config ClientConfig) (Azure, error) {
 	var err error
+	var credential azcore.TokenCredential
 
 	// Ensure we got a location
 	if config.Location == "" {
 		return nil, errors.New("location must be specified in the config")
 	}
 
-	credential, err := azidentity.NewClientSecretCredential(config.TenantID, config.ClientID, config.ClientSecret, nil)
-	if err != nil {
-		return nil, err
+	if config.ClientSecret == "" {
+		// use managed identity credentials
+		opts := azidentity.ManagedIdentityCredentialOptions{ID: azidentity.ClientID(config.ClientID)}
```

Comment:
in this case, `ClientID` is actually `userAssignedIdentityID` in azure.json?

Reply:
Yes, they are the same

Reply:
we could keep this change if end user only provides userAssignedIdentityID in backend secret, and for best convenience, we should leverage `/etc/kubernetes/azure.json` to get `userAssignedIdentityID` since it's already there on every AKS cluster.